csrc/quantization/cutlass_w8a8/scaled_mm_c3x.cu

```diff
@@ -603,6 +660,172 @@ void cutlass_gemm_sm90_fp8_dispatch(torch::Tensor& out, torch::Tensor const& a,
+               (M > 4032 && M <= 4096) || (M > 5312 && M <= 5376) ||
+               (M > 5504 && M <= 5632) || (M > 5824 && M <= 5888) ||
+               (M > 7616 && M <= 7680) || (M > 8064 && M <= 32768)) {
+      // Gemm_3
+      return cutlass_gemm_caller<Cutlass3xGemmP2>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    } else if ((M > 256 && M <= 320) || (M > 512 && M <= 576) ||
+               (M > 768 && M <= 896) || (M > 1024 && M <= 1152) ||
+               (M > 1280 && M <= 1344) || (M > 1408 && M <= 1472) ||
+               (M > 1536 && M <= 1600) || (M > 1792 && M <= 1856) ||
+               (M > 2560 && M <= 2624) || (M > 3328 && M <= 3392)) {
+      // Gemm_4
+      return cutlass_gemm_caller<Cutlass3xGemmP3>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    } else if ((M > 576 && M <= 640) || (M > 1344 && M <= 1408) ||
+               (M > 1600 && M <= 1728) || (M > 1856 && M <= 1984) ||
+               (M > 2048 && M <= 2240) || (M > 2304 && M <= 2560) ||
+               (M > 2624 && M <= 2752) || (M > 2816 && M <= 3328) ||
+               (M > 3392 && M <= 3520) || (M > 3584 && M <= 3776) ||
+               (M > 3840 && M <= 4032) || (M > 4096 && M <= 5312) ||
+               (M > 5376 && M <= 5504) || (M > 5632 && M <= 5824) ||
+               (M > 5888 && M <= 7616) || (M > 7680 && M <= 8064)) {
+      // Gemm_5
+      return cutlass_gemm_caller<Cutlass3xGemmP3>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    }
+
+  }  // end of N = 8192, K = 28672
+
+  // N = 8192, K = 8192
+  else if (N == 8192 && K == 8192) {
+    if ((M > 0 && M <= 64) || (M > 256 && M <= 320)) {
+      // Gemm_1
+      return cutlass_gemm_caller<Cutlass3xGemmP0>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    } else if ((M > 64 && M <= 128) || (M > 320 && M <= 384)) {
+      // Gemm_2
+      return cutlass_gemm_caller<Cutlass3xGemmP1>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    } else if ((M > 128 && M <= 256) || (M > 384 && M <= 512) ||
+               (M > 640 && M <= 768) || (M > 896 && M <= 1024) ||
+               (M > 1152 && M <= 1280) || (M > 1472 && M <= 1536) ||
+               (M > 1664 && M <= 1792) || (M > 5568 && M <= 5632) ||
+               (M > 5824 && M <= 5888) || (M > 7552 && M <= 7680) ||
+               (M > 7872 && M <= 32768)) {
+      // Gemm_3
+      return cutlass_gemm_caller<Cutlass3xGemmP2>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    } else if ((M > 512 && M <= 640) || (M > 768 && M <= 896) ||
+               (M > 1024 && M <= 1152) || (M > 1280 && M <= 1472) ||
+               (M > 1536 && M <= 1664) || (M > 1792 && M <= 5568) ||
+               (M > 5632 && M <= 5824) || (M > 5888 && M <= 7552) ||
+               (M > 7680 && M <= 7872)) {
+      // Gemm_4
+      return cutlass_gemm_caller<Cutlass3xGemmP3>(
+          out, a, b, std::forward<EpilogueArgs>(args)...);
+    }
+  }  // end of N = 8192, K = 8192
```

Comment:
Can you move this logic to a function in a .cuh file (something like https://github.com/vllm-project/vllm/blob/main/csrc/quantization/cutlass_w8a8/scaled_mm_c2x_sm80_dispatch.cuh) and then call the that function based on N and K values ? 

I believe it'll be better of maintainability and the logic in this function will be clearer.